### ISSUE TITLE
CI does not build the MPS on Windows

### ISSUE THREAD

**Issue by rptb1** [Ravenbrook/mps#103]
Windows is a target platform for the MPS. https://github.com/Ravenbrook/mps/blob/5051593b21c80f12d30437ed4eccf197988db98c/readme.txt#L67-L68

Travis CI didn't support Windows until fairly recently.

I enabled Windows builds in https://github.com/Ravenbrook/mps/commit/b4592dfbc2548a5296b259d7dc744915fe3f48b8 https://github.com/Ravenbrook/mps/commit/6730d24065356e0da18d4a997db36894c7041df5 https://github.com/Ravenbrook/mps/commit/e78c6e16735d7f16ef86a7f2f8356791a18c8a6e but I did so directly on the masters.  This was wrong of me.

Now we don't have a branch that we can easily merge in to version/1.117.

We should create such a branch retrospectively so that we have proper records and can do that merge.

References
- Travis CI, [The Windows Build Environment](https://docs.travis-ci.com/user/reference/windows/)

**Comment by rptb1** [comment]
Created branch as it should have been, by
```
rb@plover:~/git/mps$ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'github/master'.
rb@plover:~/git/mps$ git log --oneline -n 100 | grep -C 3 'enabling Window'
01f75231a Adding pull request merge procedure.
e78c6e167 Moving Windows outside the matrix so we don't get extraneous builds for non-existent platforms like W3I6GC that try to use configure.
6730d2406 Fix Travis CI build for Windows using auxiliary batch script to escape from Git Bash to Visual Studio.
b4592dfbc Experimentally enabling Window build with Travis CI to see what happens, since the documentation <https://docs.travis-ci.com/user/reference/windows/> is very vague.
73cc7480a Avoid compilation error from Clang 13.1.
5f175218d Merge pull request #29 from branch/2020-08-29/page-sparering-elim
555376302 Handle partial allocation failure by marking allocated pages spare and purging, as suggested by RB in review.
rb@plover:~/git/mps$ git branch branch/2023-01-05/travis-windows 73cc7480a
rb@plover:~/git/mps$ git checkout branch/2023-01-05/travis-windows
Switched to branch 'branch/2023-01-05/travis-windows'
rb@plover:~/git/mps$ git merge --ff-only e78c6e167
Updating 73cc7480a..e78c6e167
Fast-forward
 .travis.yml     |  8 +++++++-
 code/w3i6mv.bat | 10 ++++++++++
 2 files changed, 17 insertions(+), 1 deletion(-)
 create mode 100644 code/w3i6mv.bat
rb@plover:~/git/mps$ git log --oneline -n 5
e78c6e167 (HEAD -> branch/2023-01-05/travis-windows, github/branch/2023-01-06/import-project-docs, branch/2023-01-06/import-project-docs) Moving Windows outside the matrix so we don't get extraneous builds for non-existent platforms like W3I6GC that try to use configure.
6730d2406 Fix Travis CI build for Windows using auxiliary batch script to escape from Git Bash to Visual Studio.
b4592dfbc Experimentally enabling Window build with Travis CI to see what happens, since the documentation <https://docs.travis-ci.com/user/reference/windows/> is very vague.
73cc7480a (perforce/branch/2022-08-21/compile, github/branch/2022-08-21/compile, refs/pull/github/79) Avoid compilation error from Clang 13.1.
5f175218d Merge pull request #29 from branch/2020-08-29/page-sparering-elim
rb@plover:~/git/mps$ git push github
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: 
remote: Create a pull request for 'branch/2023-01-05/travis-windows' on GitHub by visiting:        
remote:      https://github.com/Ravenbrook/mps/pull/new/branch/2023-01-05/travis-windows        
remote: 
To github.com:Ravenbrook/mps.git
 * [new branch]          branch/2023-01-05/travis-windows -> branch/2023-01-05/travis-windows
```

**Comment by rptb1** [comment]
The consensus (and advice of @gareth-rees ) is cherry pick.  We want to retain some sort of traceability, so we use `-x`.  We can cherry-pick a branch like this:
```
git cherry-pick -x $(git merge-base master branch/2023-01-10/travis-freebsd)..branch/2023-01-10/travis-freebsd
```
It's pretty bad.

**Comment by rptb1** [comment]
See also @gareth-rees e-mail "[Challenges for MPS development](https://info.ravenbrook.com/mail/2021/08/08/14-26-07/0/)" section 1.2, Windows continuous integration.

@gareth-rees notes that Ravenbrook's in-house Jenkins setup did CI against our commercial clients build and test.  We can't do that on public CI.

**Comment by thejayps** [comment]
Resolved by #107
